handlers/handlers.go

```diff
@@ -195,11 +212,17 @@ func HandleSessionRequest(
 	}
 }
 
+// AuthentictedRequestHandlerType is the function signature for all authenticated
+// handlers
 type AuthentictedRequestHandlerType func(
 	http.ResponseWriter,
 	*http.Request,
 	models.UserId)
 
+// AuthenticateOrRedirectToLogin is a meta handler that resonds to all request.
+// It tries to authenticate the input request. If it's succesfull it call the passed
+// in authenticatedHandlerFunc. On failures to authenticate it redirects to the
```

Comment:
`On failures to authenticate` sounds weird. Perhaps `When it fails to authenticate, ...`